### DISCUSSION TITLE
Force Double Meta Tags (duplicated meta tags)

### DISCUSSION THREAD

**Discussion by Adrianjs42** [vercel/next.js#12493] [Help]
Is it possible to force next.js to allow duplicated meta tags?
According to [Swiftype Meta Tags](https://swiftype.com/documentation/site-search/crawler-configuration/meta-tags) I need e.g.
```
<head>
  <title>page title | website name</title>
  <meta class="swiftype" name="tags" data-type="string" content="tag1" />
  <meta class="swiftype" name="tags" data-type="string" content="tag2" />
</head>
```
What I have tried so far:

- When I inject the meta tags in a layout wrapper with next/head only the last tag (tag2) is been taken
- react-helmet works but I need a ssr approach (so it could be read by the swiftype bot)
- With dangerouslySetInnerHTML I got an error message:  https://err.sh/next.js/next-head-count-missing

Is there a better approach?

Thanks in advance!

**Comment by leerob** [comment]
You can override `_document.js` and use `head`. 

![image](https://user-images.githubusercontent.com/9113740/81095040-07958980-8eca-11ea-9bac-8fb55a09d4f0.png)


```js
import Document, { Html, Main, NextScript } from "next/document";

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx);
    return { ...initialProps };
  }

  render() {
    return (
      <Html>
        <head>
          <title>page title | website name</title>
          <meta
            class="swiftype"
            name="tags"
            data-type="string"
            content="tag1"
          />
          <meta
            class="swiftype"
            name="tags"
            data-type="string"
            content="tag2"
          />
        </head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
```

Example: https://codesandbox.io/s/nextjs-forced-meta-tags-jfrmr

**Reply by Adrianjs42** [reply]
thanks for your answer! 
Is it possible to pass props to the _document.js? 
Because right now I fetching the tags from an API and passing them via props to a layout component, so I need "dynamic tags".

**Reply by leerob** [reply]
Sure, you can use `getInitialProps` to make an API call to Swiftype. Then, you can use the props inside the `render` method. 

**Note:** This will opt you out of static optimization. 

Related: https://github.com/zeit/next.js/discussions/12490
Docs: https://nextjs.org/docs/advanced-features/custom-document

**Reply by Adrianjs42** [reply]
The problem ist that I request, based on the route, an API with a custom express server. 

pages/program/video/[id].tsx 
```
const DetailPage: React.FunctionComponent<Props> = ({ data }) => {
  if (!data) return <Loading></Loading>;
  return (
    <Layout
      tags={data && data.tags}
      ...
    </Layout>
  )
}
export async function getServerSideProps(context: any) {
  let port = process.env.NODE_ENV === 'development' ? 3000 : 80;
  const { data } = await apiFetchWrapper(
    `http://localhost:${port}/api/programm/sendung/${context.params.id}`,
  );
  if (data) {
    return {
      props: { data }, // will be passed to the page component as props
    };
  }
}
```
Layout.tsx
```
type Props = {
  tags?: any;
}
const Layout: React.FunctionComponent<Props> = ({
  tags,
}) => (
  <React.Fragment>
    <Head>
      {tags &&
        tags.map((name: string, index: any) => {
          return (
            <meta
              key={name + index}
              className="swiftype"
              name="tag"
              data-type="string"
              content={name}
            />
          );
        })}
    </Head>
    ...
  </React.Fragment>
);
export default Layout;
```
